src/app/parcels/add/databaseFunctions.ts

```diff
@@ -0,0 +1,21 @@
+import supabase, { InsertSchema, Schema } from "@/supabase";
+
+export type ParcelDatabaseInsertRecord = InsertSchema["parcels"];
+type ParcelsDatabaseFetchRecord = Pick<Schema["parcels"], "primary_key" | "client_id">;
+
+export const insertParcel = async (
+    parcelRecord: ParcelDatabaseInsertRecord
+): Promise<ParcelsDatabaseFetchRecord> => {
+    const {
+        data: ids,
+        status,
+        error,
+    } = await supabase.from("parcels").insert(parcelRecord).select("primary_key, client_id");
+
+    if (error === null && Math.floor(status / 100) === 2) {
+        return ids![0];
+    }
+    throw new Error(
+        `Error occurred whilst inserting into Parcels table. HTTP Code: ${status}, PostgreSQL Code: ${error?.code}. `
```

Comment:
Do we want to show stuff like HTTP code and Postgre Code? It might not be user friendly? Any opinions on this @TomCusack-Softwire .

Reply:
I've discussed this with Tom before and we decided on this. But he might have a change of mind. I do think this is fine for now, since we're likely going to add a new ticket soon which deals with handling errors & making an error page.